src/Bicep.LangServer.IntegrationTests/TelemetryTests.cs

```diff
@@ -313,42 +318,70 @@ public async Task BicepFileOpen_ShouldFireTelemetryEvent()
+    Application_Type: 'web'
+  }
+}
+
+module apimGroups 'groups.bicep' = if (deployGroups) {
+  name: 'apimGroups'
+}
+
+param location string = 'testLocation'";
+            var mainBicepFilePath = FileHelper.SaveResultFile(TestContext, "main.bicep", mainBicepFileContents, testOutputPath);
+            var mainUri = DocumentUri.FromFileSystemPath(mainBicepFilePath);
+            fileSystemDict[mainUri.ToUri()] = mainBicepFileContents;
+
+            var referencedBicepFileContents = @"resource parentAPIM 'Microsoft.ApiManagement/service@2019-01-01' existing = {
+  name: 'testApimInstanceName'
+}
+
+resource apimGroup 'Microsoft.ApiManagement/service/groups@2020-06-01-preview' = {
+  name: 'apiManagement/groups'
+}
+
+param storageAccount string = 'testStorageAccount'
+param location string = 'testLocation'
+
+var useDefaultSettings = true";
+            var referencedBicepFilePath = FileHelper.SaveResultFile(TestContext, "groups.bicep", referencedBicepFileContents, testOutputPath);
+            var moduleUri = DocumentUri.FromFileSystemPath(referencedBicepFilePath);
+            fileSystemDict[moduleUri.ToUri()] = referencedBicepFileContents;
+
+            var telemetryEventsListener = new MultipleMessageListener<BicepTelemetryEvent>();
 
             using var helper = await LanguageServerHelper.StartServerWithClientConnectionAsync(
                 TestContext,
                 options =>
                 {
-                    options.OnPublishDiagnostics(diags => diagsListener.AddMessage(diags));
-                    options.OnTelemetryEvent<BicepTelemetryEvent>(telemetry => telemetryReceived.SetResult(telemetry));
-                });
+                    options.OnTelemetryEvent<BicepTelemetryEvent>(telemetry => telemetryEventsListener.AddMessage(telemetry));
+                },
+                creationOptions: new Server.CreationOptions(FileResolver: new InMemoryFileResolver(fileSystemDict)));
             var client = helper.Client;
 
-            client.TextDocument.DidOpenTextDocument(TextDocumentParamHelper.CreateDidOpenDocumentParams(documentUri, bicepFileContents, 1));
-
-            var diagsParams = await diagsListener.WaitNext();
-            diagsParams.Uri.Should().Be(documentUri);
+            client.TextDocument.DidOpenTextDocument(TextDocumentParamHelper.CreateDidOpenDocumentParams(mainUri, fileSystemDict[mainUri.ToUri()], 1));
 
-            var bicepTelemetryEvent = await IntegrationTestHelper.WithTimeoutAsync(telemetryReceived.Task);
+            var bicepTelemetryEvent = await telemetryEventsListener.WaitNext();
 
             IDictionary<string, string> properties = new Dictionary<string, string>
             {
                 { "enabled", "true" },
                 { "simplify-interpolation", "warning" },
                 { "no-unused-vars", "info" },
                 { "no-hardcoded-env-urls", "warning" },
-                { "no-unused-params", "info" },
+                { "no-unused-params", "warning" },
```

Comment:
At some point (not now) these tests need to be changed so that you don't have to change them every time you add a new rule.  Ideally would be to auto-fill the list using inspection of the linter classes (and maybe manually verify that the list contains one or two expected rule IDs so we know the inspection code is working).  Thx.